zallet/src/components/sync/steps.rs

```diff
@@ -48,29 +58,53 @@ impl ChainBlock {
     }
 
     pub(super) async fn tip(chain: &FetchServiceSubscriber) -> Result<Self, SyncError> {
-        Self::resolve_inner(chain, "-1".into()).await
+        let block_id = chain.get_latest_block().await?;
+
+        let compact_block = chain.get_block(block_id).await?;
+
+        Ok(Self {
+            height: BlockHeight::from_u32(compact_block.height.try_into().map_err(
+                |e: std::num::TryFromIntError| {
+                    SyncError::from(FetchServiceError::SerializationError(e.into()))
+                },
+            )?),
+            hash: BlockHash::try_from_slice(compact_block.hash.as_slice())
+                .expect("block hash missing"),
+            prev_hash: BlockHash::try_from_slice(compact_block.prev_hash.as_slice()),
+        })
     }
 
     async fn resolve_inner(
         chain: &FetchServiceSubscriber,
         hash_or_height: String,
     ) -> Result<Self, SyncError> {
-        // TODO: https://github.com/zingolabs/zaino/issues/249
-        match chain.fetcher.get_block(hash_or_height, None).await? {
-            GetBlockResponse::Raw(_) => unreachable!("We requested verbosity 1"),
-            GetBlockResponse::Object {
-                hash,
-                height,
-                previous_block_hash,
-                ..
-            } => Ok(Self {
-                height: height
-                    .map(|h| BlockHeight::from_u32(h.0))
-                    .unwrap_or_else(|| todo!()),
-                hash: BlockHash(hash.0 .0),
-                prev_hash: previous_block_hash.map(|h| BlockHash(h.0 .0)),
-            }),
-        }
+        let hash_or_height: HashOrHeight = hash_or_height
+            .parse()
+            .map_err(|e| SyncError::from(FetchServiceError::SerializationError(e)))?;
+
+        let block_id = match hash_or_height {
+            HashOrHeight::Hash(hash) => zaino_proto::proto::service::BlockId {
+                height: 0,
+                hash: hash.0.to_vec(),
+            },
+            HashOrHeight::Height(height) => zaino_proto::proto::service::BlockId {
+                height: height.0 as u64,
+                hash: Vec::new(),
+            },
+        };
+
+        let compact_block = chain.get_block(block_id).await?;
```

Comment:
@idky137 you switched to using this API in multiple places in this PR. After several hours of debugging issues post-merge, I've discovered that this API is incomplete. It assumes that `BlockId.height` is always set, and never checks for the presence of `BlockId.hash`. This is a problem because in gRPC, not setting a field is equivalent to its default value, which means that every by-hash block query is now instead attempting to fetch the genesis block, which then fails somewhere in the depths of either Zaino or the backing full node because something is expecting `fOverwintered` to be set (which is not the case for any genesis block).

Reply:
Opened https://github.com/zingolabs/zaino/issues/254.